test/e2e/serial/tests/configuration.go

```diff
@@ -343,5 +345,137 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
 			Expect(err).ToNot(HaveOccurred())
 			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod.Namespace, updatedPod.Name, schedulerTestName)
 		})
+
+		It("[test_id:47585][reboot_required] can change kubeletconfig and controller should adapt", func() {
+			nroOperObj := &nropv1alpha1.NUMAResourcesOperator{}
+			err := fxt.Client.Get(context.TODO(), client.ObjectKey{Name: objects.NROName()}, nroOperObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", objects.NROName())
+
+			initialNrtList := nrtv1alpha1.NodeResourceTopologyList{}
+			initialNrtList, err = e2enrt.GetUpdated(fxt.Client, initialNrtList, timeout)
+			Expect(err).ToNot(HaveOccurred(), "cannot get any NodeResourceTopology object from the cluster")
+
+			mcps, err := nropmcp.GetNodeGroupsMCPs(context.TODO(), fxt.Client, nroOperObj.Spec.NodeGroups)
+			Expect(err).ToNot(HaveOccurred(), "cannot get MCPs associated with NUMAResourcesOperator %q", nroOperObj.Name)
+
+			kcList := &machineconfigv1.KubeletConfigList{}
+			err = fxt.Client.List(context.TODO(), kcList)
+			Expect(err).ToNot(HaveOccurred())
+
+			targetedKCs := make([]*machineconfigv1.KubeletConfig, 0)
+			for _, mcp := range mcps {
+				for i := 0; i < len(kcList.Items); i++ {
+					kc := &kcList.Items[i]
+					kcMcpSel, err := metav1.LabelSelectorAsSelector(kc.Spec.MachineConfigPoolSelector)
+					Expect(err).ToNot(HaveOccurred())
+
+					if kcMcpSel.Matches(labels.Set(mcp.Labels)) {
+						targetedKCs = append(targetedKCs, kc)
+					}
+				}
+			}
+			Expect(len(targetedKCs)).ToNot(BeZero(), "there should be at least one kubeletconfig.machineconfiguration object")
+
+			By("modifying reserved CPUs under kubeletconfig")
+			initialRsvCPUs := make(map[string]string)
+			for i := 0; i < len(targetedKCs); i++ {
```

Comment:
do we want to change all the kubeletconfig we find or just one?
meaning, what's the upside of changing all of them?

Reply:
We probably can pick one randomly and wait for the relevant MCPs.
But in this specific setup, I presume there will be only one so it doesn't really matter.

Reply:
true, when running in our CI. But we want to make preparations for the test to be _possibly_ used as validation tool. So let's not make too many implicit assumptions over the environment, and let's be explicit about test requirements and scenarios